tests/store/artifact/test_optimized_s3_artifact_repo.py

```diff
@@ -98,12 +102,29 @@ def test_get_s3_client_verify_param_set_correctly(
         )
 
 
-def test_get_s3_client_region_name_set_correctly(s3_artifact_root):
-    region_name = "us_random_region_42"
+@pytest.mark.parametrize("client_throws", [True, False])
+def test_get_s3_client_region_name_set_correctly(s3_artifact_root, client_throws):
+    if client_throws:
```

Comment:
Nice, maybe just use the same region name in both test cases?

Reply:
In these grid tests, esp with so much opaqueness as to what's happening under the hood, I'd like to make sure that the strings are unique across tests?

Reply:
Thankfully pytest will run each test individually as a separate test case, so we should be able to single out the failing test by name vs having to change values in each test case

Reply:
(it might be helpful to have separate names if we were running a single test that iterates over parameters in a loop, but `pytest.mark.parametrize` generates multiple test cases, which eliminates the need for extra mangling ourselves)